### PR TITLE
Smoketests bazel

### PR DESCRIPTION
**Pull request by drewmacrae**

### CODE REVIEW

**Comment by drewmacrae** [comment]
I still have a lot of cleanup to do here, but it's working for some of the smoketests so far.

**Comment by drewmacrae** [comment]
I need to take a look at the OTBN work here: https://github.com/lowRISC/opentitan/pull/9412 and get my head around it so I can get that building.

**Comment by drewmacrae** [comment]
The OTBN and OTP tests are now building and passing. I still need to fix the CSRNG and the GPIO tests

File: sw/device/tests/BUILD [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+        "//sw/device/lib/dif:spi_device",
+        "//sw/device/lib/dif:uart",
+        "//sw/device/lib/runtime:hart",
+        "//sw/device/lib/runtime:log",
+        "//sw/device/lib/runtime:print",
+        "//sw/device/lib/testing:entropy_testutils",
+        "//sw/device/lib/testing/test_framework",
+        "//sw/device/lib/testing/test_framework:ottf",
+    ],
+)
```
Comment (cfrantz) [inline sw/device/tests/BUILD:37]: Is there a reason to structure these as separate targets?  This 'lib' doesn't have any srcs of its own, so its  just a list of deps for the test program and quite a few of these deps are automatically part of the `opentitan_functest` rule.

File: sw/device/tests/BUILD [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+        "//sw/device/lib/arch:device",
+        "//sw/device/lib/crt",
+        "//sw/device/lib/dif:clkmgr",
+        "//sw/device/lib/dif:gpio",
+        "//sw/device/lib/dif:spi_device",
+        "//sw/device/lib/dif:uart",
+        "//sw/device/lib/runtime:hart",
+        "//sw/device/lib/runtime:log",
+        "//sw/device/lib/runtime:print",
+        #"//sw/device/lib/testing:entropy_testutils",
```
Comment (cfrantz) [inline sw/device/tests/BUILD:95]: copy-pasta? (here and below)

**Review by cfrantz — COMMENTED** [review]
Overall, this looks p[retty good to me.

File: sw/device/tests/BUILD (hunk 1 above)
Comment (drewmacrae) [reply sw/device/tests/BUILD]: I didn't know that about the `opentitan_functest` rule, and hadn't yet worked out the organizing principle for when we do or don't split things. You're right that combining them simplifies this PR, and there's no reason for splitting them.

File: sw/device/tests/BUILD (hunk 2 above)
Comment (drewmacrae) [reply sw/device/tests/BUILD]: yes

**Review by mcy — APPROVED** [review]

**Pull request merged**